### PR TITLE
Allow NO_GENERATED_KEYS option for PreparedStatement 

### LINKED ISSUE #22100: PreparedStatement dont support java.sql.Statement.NO_GENERATED_KEYS option
Hi all,

I just noticed that it is not possible to execute the `java.sql.Connection.prepareStatement(String sql, int option)` method with option on `java.sql.Statement.NO_GENERATED_KEYS` without getting the error:

> java.sql.SQLFeatureNotSupportedException: Auto generated keys must be NO_GENERATED_KEYS

It is still possible to call this method without an option. But using the option `java.sql.Statement.NO_GENERATED_KEYS` should be the same as without the option.

### PR DESCRIPTION
**Pull request by prrvchr**
## Description

adds support for PreparedStatement creation with the 'NO_GENERATED_KEYS' option

Fixes #22100

## Release notes

```markdown
# JDBC driver
* Allow only `NO_GENERATED_KEYS` option in `Connection.prepareStatement` method. ({issue}`22100`)
```

### CODE REVIEW

File: client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoConnection.java [hunk 1]
```diff
@@ -469,7 +469,7 @@ public CallableStatement prepareCall(String sql, int resultSetType, int resultSe
     public PreparedStatement prepareStatement(String sql, int autoGeneratedKeys)
             throws SQLException
     {
-        if (autoGeneratedKeys != Statement.RETURN_GENERATED_KEYS) {
+        if (autoGeneratedKeys != Statement.NO_GENERATED_KEYS) {
```
Comment (ebyhr) [inline client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoConnection.java:472]: Please add a test to `TestJdbcConnection`.

**Review by ebyhr — COMMENTED** [review]
Please capitalize and shorten the commit message. https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#commits-and-pull-requests

File: client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoConnection.java (hunk 1 above)
Comment (prrvchr) [reply client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoConnection.java:472]: If we want to test, we apparently need a `PreparedStatement` with Hive. I don't see which request to submit under Hive (I don't know Hive)?

Comment (ebyhr) [reply client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoConnection.java:472]: TestJdbcConnection already uses Hive connector.

Comment (prrvchr) [reply client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoConnection.java:472]: > TestJdbcConnection already uses Hive connector.

Comment (prrvchr) [reply client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoConnection.java:472]: @ebyhr I just added the tests

File: client/trino-jdbc/src/test/java/io/trino/jdbc/TestJdbcConnection.java [hunk 2]
```diff
@@ -465,6 +465,29 @@ public void testConcurrentCancellationOnConnectionClose()
         testConcurrentCancellationOnConnectionClose(false);
     }
 
+    @Test
```
Comment (prrvchr) [inline client/trino-jdbc/src/test/java/io/trino/jdbc/TestJdbcConnection.java:468]: Here we test if we can create a `PreparedStatement` with the options:
- `NO_GENERATED_KEYS`
- `RETURN_GENERATED_KEYS`
The content of the request allowing the creation of this `PreparedStatement` is of a minimal format.

File: client/trino-jdbc/src/test/java/io/trino/jdbc/TestJdbcConnection.java [hunk 3]
```diff
@@ -465,6 +465,29 @@ public void testConcurrentCancellationOnConnectionClose()
...
+    public void testPreparedStatementCreationOption()
+            throws SQLException
+    {
+        try (Connection connection = createConnection()) {
+            createPreparedStatementWithOption(connection, Statement.NO_GENERATED_KEYS, true);
+            createPreparedStatementWithOption(connection, Statement.RETURN_GENERATED_KEYS, false);
+        }
+    }
+
+    private void createPreparedStatementWithOption(Connection connection, int option, boolean support)
```
Comment (ebyhr) [inline client/trino-jdbc/src/test/java/io/trino/jdbc/TestJdbcConnection.java:478]: Please inline this method and use `assertThatThrownBy` instead.

**Review by ebyhr — COMMENTED** [review]
Please squash commits into one and change the commit title to "Allow NO_GENERATED_KEYS option for PreparedStatement"

File: client/trino-jdbc/src/test/java/io/trino/jdbc/TestJdbcConnection.java [hunk 4]
```diff
@@ -465,6 +467,21 @@ public void testConcurrentCancellationOnConnectionClose()
         testConcurrentCancellationOnConnectionClose(false);
     }
 
+    @Test
```
Comment (ebyhr) [inline client/trino-jdbc/src/test/java/io/trino/jdbc/TestJdbcConnection.java:470]: Please move under `testSessionUser` method. The current place is located between `testConcurrentCancellationOnConnectionClose ()` and the helper method `testConcurrentCancellationOnConnectionClose(boolean autoCommit)`.

Comment (prrvchr) [reply client/trino-jdbc/src/test/java/io/trino/jdbc/TestJdbcConnection.java:454]: Done

**Comment by prrvchr** [comment]
@ebyhr , I would like to thank you...

**Pull request merged**